block/src/Pos/Block/Pure.hs

```diff
@@ -59,14 +59,19 @@ data VerifyHeaderParams = VerifyHeaderParams
       -- ^ Check that header has no unknown attributes.
     } deriving (Eq, Show)
 
+verifyFromEither :: Text -> Either a b -> VerificationRes
+verifyFromEither txt (Left _)  = verifyGeneric [(False, txt)]
```

Comment:
I don't like it, because in many cases this `Left` contains useful information (error message) which gets lost by this function.

Reply:
Fixed in CSL-2072, can be skipped here.

Reply:
What should I do with the second `Text`? Is `verifyGeneric [(False, txt), (False, otherTxt)]` the right thing to do?

Reply:
See https://github.com/input-output-hk/cardano-sl/pull/2380/files#diff-f5f4326d089f2fe7ec58ac61170f883cR64